### DISCUSSION TITLE
How to deal with microservices in individual repositories?

### DISCUSSION THREAD

**Discussion by cmenzi** [microsoft/aspire#1060] [Q&A] [answered]
Are there any options to use Aspire with a microservices spread over multiple repositories? Or do we need first migrate everything into a monorepo?

We heavily focus that our teams can develop, deploy and operate independently. I also don't want that everything becomes a single deployment unit.

Is it good idea to use Aspire with `builder.AddContainer(...)` to reference another microservice? What functionality do we loose with this approach in regards of health-checks, instrumentation, ...? How does the deployment look like, when those microservice containers should not be used as sidecar in production?

I would be really great if there would be some documentation to show best practices for such scenarios.

**Comment by davidfowl** [comment, marked as answer]
Follow the discussion here https://github.com/dotnet/aspire/issues/829
